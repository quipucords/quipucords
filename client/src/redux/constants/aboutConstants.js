const ABOUT_DIALOG_OPEN = 'ABOUT_DIALOG_OPEN';
const ABOUT_DIALOG_CLOSE = 'ABOUT_DIALOG_CLOSE';

export { ABOUT_DIALOG_OPEN, ABOUT_DIALOG_CLOSE };
