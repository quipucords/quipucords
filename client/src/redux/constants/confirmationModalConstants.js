const CONFIRMATION_MODAL_SHOW = 'CONFIRMATION_MODAL_SHOW';
const CONFIRMATION_MODAL_HIDE = 'CONFIRMATION_MODAL_HIDE';

export { CONFIRMATION_MODAL_SHOW, CONFIRMATION_MODAL_HIDE };
