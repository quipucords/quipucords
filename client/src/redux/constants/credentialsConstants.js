
const LOAD_CREDENTIALS_SUCCESS = 'LOAD_CREDENTIALS_SUCCESS';

export {
  LOAD_CREDENTIALS_SUCCESS
}
