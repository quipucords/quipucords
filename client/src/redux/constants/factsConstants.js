const ADD_FACTS_ERROR = 'ADD_FACTS_ERROR';
const ADD_FACTS_LOADING = 'ADD_FACTS_LOADING';
const ADD_FACTS_SUCCESS = 'ADD_FACTS_SUCCESS';

export { ADD_FACTS_ERROR, ADD_FACTS_LOADING, ADD_FACTS_SUCCESS };
