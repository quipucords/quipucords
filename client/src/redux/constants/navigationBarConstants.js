const NAV_TOGGLE_COLLAPSE = 'NAV_TOGGLE_COLLAPSE';
const NAV_COLLAPSE = 'NAV_COLLAPSE';
const NAV_OPEN = 'NAV_OPEN';

export { NAV_TOGGLE_COLLAPSE, NAV_COLLAPSE, NAV_OPEN };
