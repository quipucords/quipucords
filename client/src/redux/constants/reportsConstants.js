const GET_REPORT = 'GET_REPORT';
const GET_MERGE_REPORT = 'GET_MERGE_REPORT';

export { GET_REPORT, GET_MERGE_REPORT };
