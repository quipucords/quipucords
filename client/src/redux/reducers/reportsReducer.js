import helpers from '../../common/helpers';
import { reportsTypes } from '../constants';

const initialState = {
  report: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false,
    reports: []
  },

  merge: {
    error: false,
    errorMessage: '',
    pending: false,
    fulfilled: false
  }
};

const reportsReducer = function(state = initialState, action) {
  switch (action.type) {
    // Error/Rejected
    case helpers.REJECTED_ACTION(reportsTypes.GET_REPORT):
      return helpers.setStateProp(
        'report',
        {
          error: action.error,
          errorMessage: helpers.getErrorMessageFromResults(action.payload)
        },
        {
          state,
          initialState
        }
      );

    case helpers.REJECTED_ACTION(reportsTypes.GET_MERGE_REPORT):
      return helpers.setStateProp(
        'merge',
        {
          error: action.error,
          errorMessage: helpers.getErrorMessageFromResults(action.payload)
        },
        {
          state,
          initialState
        }
      );

    // Loading/Pending
    case helpers.PENDING_ACTION(reportsTypes.GET_REPORT):
      return helpers.setStateProp(
        'report',
        {
          pending: true
        },
        {
          state,
          initialState
        }
      );

    case helpers.PENDING_ACTION(reportsTypes.GET_MERGE_REPORT):
      return helpers.setStateProp(
        'merge',
        {
          pending: true
        },
        {
          state,
          initialState
        }
      );

    // Success/Fulfilled
    case helpers.FULFILLED_ACTION(reportsTypes.GET_REPORT):
      return helpers.setStateProp(
        'report',
        {
          fulfilled: true,
          reports: action.payload.data
        },
        {
          state,
          initialState
        }
      );

    case helpers.FULFILLED_ACTION(reportsTypes.GET_MERGE_REPORT):
      return helpers.setStateProp(
        'merge',
        {
          fulfilled: true
        },
        {
          state,
          initialState
        }
      );

    default:
      return state;
  }
};

reportsReducer.initialState = initialState;

export { initialState, reportsReducer };

export default reportsReducer;
