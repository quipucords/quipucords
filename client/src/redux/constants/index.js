import * as aboutTypes from './aboutConstants';
import * as credentialsTypes from './credentialsConstants';
import * as factsTypes from './factsConstants';
import * as navigationBarTypes from './navigationBarConstants';
import * as reportsTypes from './reportsConstants';
import * as scansTypes from './scansConstants';
import * as sourcesTypes from './sourcesConstants';
import * as viewToolbarTypes from './viewToolbarConstants';

const reduxTypes = {
  about: aboutTypes,
  credentials: credentialsTypes,
  facts: factsTypes,
  navigation: navigationBarTypes,
  reports: reportsTypes,
  scans: scansTypes,
  sources: sourcesTypes,
  viewToolbar: viewToolbarTypes
};

export {
  reduxTypes,
  aboutTypes,
  credentialsTypes,
  factsTypes,
  navigationBarTypes,
  reportsTypes,
  scansTypes,
  sourcesTypes,
  viewToolbarTypes
};

export default reduxTypes;
