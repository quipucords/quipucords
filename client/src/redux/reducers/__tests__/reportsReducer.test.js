import reportsReducer from '../reportsReducer';
import { reportsTypes as types } from '../../constants';
import helpers from '../../../common/helpers';

describe('ReportsReducer', () => {
  it('should return the initial state', () => {
    expect(reportsReducer.initialState).toBeDefined();
  });

  it('should handle all defined error types', () => {
    const specificTypes = [types.GET_REPORT, types.GET_MERGE_REPORT];

    specificTypes.forEach(value => {
      const dispatched = {
        type: helpers.REJECTED_ACTION(value),
        error: true,
        payload: {
          message: 'MESSAGE',
          response: {
            status: 0,
            statusText: 'ERROR TEST',
            data: {
              detail: 'ERROR'
            }
          }
        }
      };

      const resultState = reportsReducer(undefined, dispatched);

      expect({ type: helpers.REJECTED_ACTION(value), result: resultState }).toMatchSnapshot(`rejected types ${value}`);
    });
  });

  it('should handle all defined pending types', () => {
    const specificTypes = [types.GET_REPORT, types.GET_MERGE_REPORT];

    specificTypes.forEach(value => {
      const dispatched = {
        type: helpers.PENDING_ACTION(value)
      };

      const resultState = reportsReducer(undefined, dispatched);

      expect({ type: helpers.PENDING_ACTION(value), result: resultState }).toMatchSnapshot(`pending types ${value}`);
    });
  });

  it('should handle all defined fulfilled types', () => {
    const specificTypes = [types.GET_REPORT, types.GET_MERGE_REPORT];

    specificTypes.forEach(value => {
      const dispatched = {
        type: helpers.FULFILLED_ACTION(value),
        payload: {
          data: {
            test: 'success'
          }
        }
      };

      const resultState = reportsReducer(undefined, dispatched);

      expect({ type: helpers.FULFILLED_ACTION(value), result: resultState }).toMatchSnapshot(
        `fulfilled types ${value}`
      );
    });
  });
});
