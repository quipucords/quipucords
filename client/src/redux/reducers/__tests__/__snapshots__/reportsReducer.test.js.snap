// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fulfilled types GET_MERGE_REPORT 1`] = `
Object {
  "result": Object {
    "merge": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
    },
    "report": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "reports": Array [],
    },
  },
  "type": "GET_MERGE_REPORT_FULFILLED",
}
`;

exports[`fulfilled types GET_REPORT 1`] = `
Object {
  "result": Object {
    "merge": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
    "report": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": true,
      "pending": false,
      "reports": Object {
        "test": "success",
      },
    },
  },
  "type": "GET_REPORT_FULFILLED",
}
`;

exports[`pending types GET_MERGE_REPORT 1`] = `
Object {
  "result": Object {
    "merge": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
    },
    "report": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "reports": Array [],
    },
  },
  "type": "GET_MERGE_REPORT_PENDING",
}
`;

exports[`pending types GET_REPORT 1`] = `
Object {
  "result": Object {
    "merge": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
    "report": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": true,
      "reports": Array [],
    },
  },
  "type": "GET_REPORT_PENDING",
}
`;

exports[`rejected types GET_MERGE_REPORT 1`] = `
Object {
  "result": Object {
    "merge": Object {
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "pending": false,
    },
    "report": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
      "reports": Array [],
    },
  },
  "type": "GET_MERGE_REPORT_REJECTED",
}
`;

exports[`rejected types GET_REPORT 1`] = `
Object {
  "result": Object {
    "merge": Object {
      "error": false,
      "errorMessage": "",
      "fulfilled": false,
      "pending": false,
    },
    "report": Object {
      "error": true,
      "errorMessage": "ERROR",
      "fulfilled": false,
      "pending": false,
      "reports": Array [],
    },
  },
  "type": "GET_REPORT_REJECTED",
}
`;
